client/src/statistics/SelectComponent.js

```diff
@@ -0,0 +1,194 @@
+  { key: "biodiesel", text: "Biodiesel" },
+  { key: "e85", text: "E85" },
+  { key: "cng", text: "CNG" },
+  { key: "propane", text: "Propane" },
+  { key: "hydrogen", text: "Hydrogen" },
+  { key: "gasoline", text: "Gasoline" },
+  { key: "diesel", text: "Diesel" },
+];
+
+const portOptions = [
+  { key: "All", text: "All ports" },
+  { key: "type1", text: "Type-1" },
+  { key: "type2", text: "Type-2" },
+  { key: "ccs", text: "Ccs" },
+  { key: "chademo", text: "Chademo" },
+  { key: "tesla", text: "Tesla" },
+  { key: "nema515", text: "NEMA-515" },
+  { key: "nema520", text: "NEMA-520" },
+  { key: "nema1450", text: "NEMA-1450" },
+];
+
+const speedOptions = [
+  { key: "All", text: "All speeds" },
+  { key: "acLevel1", text: "AC level-1" },
+  { key: "acLevel2", text: "AC level-2" },
+  { key: "dcfast", text: "DC fast charging" },
+];
+
+const network = [
+  "Non-Networked",
+  "Volta",
+  "EV Connect",
+  "POWERFLEX",
+  "ChargePoint Network",
+  "OpConnect",
+  "SHELL_RECHARGE",
+  "EVGATEWAY",
+  "eVgo Network",
+  "AMPUP",
+  "Webasto",
+  "SemaCharge Network",
+  "UNIVERSAL",
+  "EVCS",
+  "Blink Network",
+  "FCN",
+  "Tesla",
+  "Tesla Destination",
+  "EVRANGE",
+  "Electrify America",
+  "CHARGELAB",
+  "LIVINGSTON",
+  "FLO",
+  "ZEFNET",
+  "FPLEV",
+  "RIVIAN_WAYPOINTS",
+  "RED_E",
+  "SWTCH",
+  "CIRCLE_K",
+  "WAVE",
+  "GRAVITI_ENERGY",
+  "FLASH",
+  "RIVIAN_ADVENTURE",
+  "CHARGEUP",
+];
+// map an array to dictionary
+// const networkOptions = Object.fromEntries(network.map(x => {"key":x, "value":x}));
+const networkOptions = [];
+networkOptions[0] = { key: "All", text: "All networks" };
+
+function arrToDict(item, index) {
+  networkOptions[index + 1] = { key: item, text: item };
+}
+// Sort by string order to make sure the plotting order is consistent.
+network.sort().forEach(arrToDict);
+
+function SelectComponent(type, label, defaultValue, handler, stateVariable) {
+  // use theme for this page
+  const theme = useTheme();
+  // loaded menu options based on given type
+  let obj = [];
+  if (type === "port") {
+    obj = portOptions;
+  } else if (type === "state") {
+    obj = stateOptions;
+  } else if (type === "stationType") {
+    obj = stationTypeOptions;
+  } else if (type === "vehicleType") {
+    obj = vehicleTypeOptions;
+  } else if (type === "speed") {
+    obj = speedOptions;
+  } else if (type === "network") {
+    obj = networkOptions;
+  }
+  return (
+    <FormControl sx={{ m: 1, minWidth: 300 }} size="small">
+      <InputLabel>{label}</InputLabel>
+      <Select
+        // labelId="demo-select-small"
+        // id="demo-select-small"
+        defaultValue={defaultValue}
+        value={stateVariable}
+        label={label}
+        onChange={handler}
+        sx={{
+          backgroundColor: theme.palette.background.default,
+          color: theme.palette.primary.main,
+        }}
+      >
+        {obj.map((row) => (
+          <MenuItem value={row.key} color={theme.palette.background.default}>
```

Comment:
You should probably add `key={row.key}` to `<MenuItem value={row.key} color={theme.palette.background.default}>`.

If not, React complains with the following in the console:
Warning: Each child in a list should have a unique "key" prop.

For other lists (places where you used the map method), it is better to add keys.